packages/appstore/src/components/wallet-content/__tests__/wallet-transfer-block.spec.tsx

```diff
@@ -0,0 +1,56 @@
+import React from 'react';
+import { render, screen } from '@testing-library/react';
+import '@testing-library/jest-dom';
+import WalletTransferBlock from '../wallet-transfer-block';
+import { TWalletAccount } from 'Types';
+import { StoreProvider, mockStore } from '@deriv/stores';
+
+const wallet_account: TWalletAccount = {
+    name: 'USD',
+    currency: 'USD',
+    icon: '',
+    balance: 10415.24,
+    icon_type: 'fiat',
+    landing_company_name: 'svg',
+    is_disabled: false,
+    is_virtual: false,
+    loginid: 'CRW10001',
+};
+
+jest.mock('./../../containers/currency-switcher-container', () => jest.fn(({ children }) => <div>{children}</div>));
+
+const mockedRootStore = mockStore({
+    modules: {
+        cfd: {
+            toggleCompareAccountsModal: jest.fn(),
+        },
+    },
+});
+
+describe('<WalletTransferBlock />', () => {
+    it('Check balance', () => {
+        render(
+            <StoreProvider store={mockedRootStore}>
+                <WalletTransferBlock wallet_account={wallet_account} />
+            </StoreProvider>
+        );
+        const { currency } = wallet_account;
+
+        const balance_title = screen.queryByText(`10,415.24 ${currency}`);
+
+        expect(balance_title).toBeInTheDocument();
+    });
+
+    it('Check loginid', () => {
+        render(
+            <StoreProvider store={mockedRootStore}>
+                <WalletTransferBlock wallet_account={wallet_account} />
+            </StoreProvider>
+        );
+        const { loginid } = wallet_account;
+
+        const loginid_title = screen.queryByText(String(loginid));
+
+        expect(loginid_title).toBeInTheDocument();
```

Comment:
same here. Also do we need to parse the loginid value as String? if we remove it, would the test fail?

Reply:
Since loginid is an optional value and getByText or queryByText doesn't accept undefined.